pyvisa_py/gpib.py

```diff
@@ -9,20 +9,78 @@
 
 import ctypes  # Used for missing bindings not ideal
 from bisect import bisect
-from typing import Any, Iterator, List, Tuple, Union
+from typing import Any, Iterator, List, Tuple, Type, Union
 
 from pyvisa import attributes, constants
 from pyvisa.constants import ResourceAttribute, StatusCode
-from pyvisa.rname import GPIBInstr, GPIBIntfc
+from pyvisa.rname import GPIBInstr, GPIBIntfc, parse_resource_name
 
+from . import prologix
 from .common import LOGGER
-from .sessions import Session, UnknownAttribute
+from .sessions import Session, UnavailableSession, UnknownAttribute, VISARMSession
+
+
+@Session.register(constants.InterfaceType.gpib, "INSTR")
+class GPIBSessionDispatch(Session):
+    """Dispatch to the proper class based on entries in prologix.BOARDS.
+
+    Uses the __new__ method to intercept the creation of the instance of a
+    GPIB session.  If parsed.board is found in prologix.BOARDS, create an
+    instance of prologix.PrologixInstrSession, otherwise create an instance
+    of GPIBSession.
+
+    """
+
+    def __new__(  # type: ignore[misc]
+        cls,
+        resource_manager_session: VISARMSession,
+        resource_name: str,
+        parsed=None,
+        open_timeout: int | None = None,
+    ) -> Session:
+        newcls: Type
+
+        if parsed is None:
+            parsed = parse_resource_name(resource_name)
+
+        if parsed.board in prologix.BOARDS:
+            newcls = prologix.PrologixInstrSession
```

Comment:
Since the boards are registered through the opening of a resource via the resource manager, I think it would make sense to store them on the library linked to the resource manager. To enable this we could modify the signature of the base Session class which we can do since it is not user facing.

Reply:
Thanks for the suggestion, it's definitely better to eliminate the module-level global.  I moved it to a class var instead (in the prologix._PrologixIntfcSession class).